App/Components/ThreadSelector.tsx

```diff
@@ -25,8 +25,26 @@ class ThreadSelector extends React.Component<ScreenProps & DispatchProps & Navig
     this.props.navigateToThread(id, name)
   }
 
+  _convertToPreview = (thread: ThreadData) => {
+    return {
+      id: thread.id,
+      name: thread.name,
+      // total number of images in the thread
+      size: thread.photos.length,
+      // just keep the top 2
+      photos: thread.photos.slice(0, 3),
+      // get a rough count of distinct users
+      userCount: thread.photos.length > 0 ? [...new Set(thread.photos.map((photo) => photo.author_id))].length : 1,
+      // latest update based on the latest item
+      updated: thread.photos.length > 0 && thread.photos[0].date ? Date.parse(thread.photos[0].date) : 0,
+      // latest peer to push to the thread
+      latestPeerId: thread.photos.length > 0 && thread.photos[0].author_id ? thread.photos[0].author_id : undefined
+    }
+  }
+
   _renderItem = (rowData: any) => {
-    const item: ThreadData = rowData.item
+    const thread: ThreadData = rowData.item
+    const item = this._convertToPreview(thread)
```

Comment:
This conversion should happen in mapStateToProps I think. And the conversion function could effectively be a "selector" and just live with the rest of the selector code.

Reply:
done